File: browser/src/app/api/models/AdminServiceType.ts
import { Trade } from '../../model/data-model';

export class AdminServiceType {
  id: number;
  name: string;
  description: string;
  imageUrl: string;
  isActive: boolean;
  labels: Array<string>;
  leadPrice: number;
  questionaryId: number;
  trades: Array<Trade>;

  constructor(name: string, description: string, imageUrl: string, isActive: boolean, labels: Array<string>, leadPrice: number, questionaryId: number, trades: Array<Trade>) {
    this.name = name;
    this.description = description;
    this.imageUrl = imageUrl;
    this.isActive = isActive;
    this.labels = labels;
    this.leadPrice = leadPrice;
    this.questionaryId = questionaryId;
    this.trades = trades;
  }
}

File: improver/src/main/java/com/improver/model/admin/AdminServiceType.java
package com.improver.model.admin;

import com.improver.entity.ServiceType;
import com.improver.model.NameIdTuple;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@NoArgsConstructor
@Accessors(chain = true)
public class AdminServiceType {
    private long id;
    private String name;
    private String description;
    private String imageUrl;
    private Boolean isActive;
    private List<String> labels;
    private int leadPrice;
    private long questionaryId;
    private List<NameIdTuple> trades;

    public AdminServiceType(ServiceType serviceType, Long questionaryId) {
        this.id = serviceType.getId();
        this.name = serviceType.getName();
        this.description = serviceType.getDescription();
        this.imageUrl = serviceType.getImageUrl();
        this.isActive = serviceType.isActive();
        this.labels = serviceType.getLabels();
        this.leadPrice = serviceType.getLeadPrice();
        this.questionaryId = questionaryId != null ? questionaryId : 0;
    }
}
